internal/controller/nimcache_controller.go

```diff
@@ -959,9 +959,8 @@ func constructPodSpec(nimCache *appsv1alpha1.NIMCache, platformType k8sutil.Orch
 				},
 			},
 			SecurityContext: &corev1.PodSecurityContext{
-				RunAsUser:    nimCache.GetUserID(),
-				FSGroup:      nimCache.GetGroupID(),
-				RunAsNonRoot: ptr.To[bool](true),
```

Comment:
Rather than removing it here, if proxy is set, then we can toggle this to `false` conditionally. That way the security config will remain same.

Reply:
Actually, the container `RunAsNonRoot` overides the one on pod. So, made the change to set  `RunAsNonRoot: false` on init container